### DISCUSSION TITLE
Incorporate ./shells

### DISCUSSION THREAD

**Discussion by blaggacao** [nrdxp/nixdev#186] [General]
**Is your feature request related to a problem? Please describe.**
I think it is time to share efforts in building _user provided dev environments_. I also think this template repo leverages the flakes model of sharing to the most elaborate extend. I would be pleased if I could use something like `github.com:user/nixflk#devShells.rust` of a trusted user as an input to my own variety of devshells.

**Describe the solution you'd like**
I would like such sharing to evolve around https://github.com/numtide/devshell to provide an advanced and consistent devshell environment.

**Describe alternatives you've considered**
none &mdash; inmature thought.

**Additional context**
- https://github.com/numtide/devshell
- https://github.com/colemickens/nixcfg/tree/main/shells (for what I mean)
- _User provided dev environments_ would be oposed to _porject provided dev environments_. It would be interesting to learn about ways how user provided dev environments can hook into / overlay / complement project provided ones. This question collaterally seeks answer here: https://github.com/numtide/devshell/issues/52
- > [a few `shell.nix`s for being able to hack on random source code trees](https://github.com/colemickens/nixcfg)

   &mdash; for a quite intriguing motivation to boost collaboration on source code through the flakes model

**Comment by nrdxp** [comment]
I like the idea of providing various shells as an output. I've had a similar idea recently. I think the simplest way to go about this would be to add a custom flake output attribute. Something like `shells`. We will probably have to write some checks as well.

**Comment by blaggacao** [comment]
I hope @zimbatm could add to the thought-noodle-soop we are cooking here. Should we add more salt? A `shells` attribute is already quite spicey. But will it be allowed once / if flakes outputes will be typed?

**Comment by nrdxp** [comment]
#54 is a step in this direction.

**Comment by blaggacao** [comment]
I try to second: https://github.com/nrdxp/nixflk/pull/55
